src/main/java/org/isf/admission/manager/AdmissionBrowserManager.java

```diff
@@ -343,6 +349,11 @@ protected void validateAdmission(Admission admission, boolean insert) throws OHS
 		}
 		if (admission.getDiseaseIn() == null) {
 			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisincannotbeempty.msg")));
+		} else {
+			List<Disease> diseases = diseaseManager.getDiseaseIpdIn().stream().filter(dis -> dis == admission.getDiseaseIn()).collect(Collectors.toList());
+			if (diseases.size() > 0) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisinisnotallowed.msg")));
+			}
```

Comment:
Sorry, I forgot to add this comment: shouldn't the filter compare obects instead of integer, like this `dis -> dis.equals(admission.getDiseaseIn())` ? 

Then, if the `diseaseIn` in the passed `admission` object IS NOT found in the `diseases = diseaseManager.getDiseaseIpdIn()` list, the `diseases` list object will be empty (not greater than zero) and so the error message would make sense, don't you think?

in short:

```suggestion
		} else {
			List<Disease> diseases = diseaseManager.getDiseaseIpdIn().stream().filter(dis -> dis.equals( admission.getDiseaseIn()).collect(Collectors.toList());
			if (diseases.size().isEmpty()) {
				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisinisnotallowed.msg")));
			}
```